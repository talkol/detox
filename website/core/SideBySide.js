const React = require("react");

export function SideBySideOption({ children }) {
  return <div>{children}</div>;
}

export default class SideBySide extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedOption: null
    };
  }

  selectOption(name) {
    console.log("FOOO");
    this.setState({
      selectedOption: name
    });
  }

  renderControls(selectedOption) {
    const options = React.Children.map(
      this.props.children,
      child => child.props.name
    );

    return (
      <div
        style={{
          borderBottom: "1px solid #333",
          marginBottom: "1em",
          marginTop: "1em",
          paddingBottom: "0.1em"
        }}
      >
        {options.map(option => (
          <a
            key={option}
            onClick={this.selectOption.bind(this, option)}
            style={{
              backgroundColor: selectedOption === option ? "#F2F2F2" : "white",
              border: "1px solid #333",
              borderRadius: "5px",
              cursor: "pointer",
              display: "inline-block",
              fontWeight: "bold",
              fontSize: "1.5em",
              padding: "0.3em",
              marginRight: "0.5em",
              userSelect: "none"
            }}
          >
            {option}
          </a>
        ))}
      </div>
    );
  }

  render() {
    const { selectedOption } = this.state;
    const children = React.Children.toArray(this.props.children);
    const selectedChild = selectedOption
      ? children.find(({ props }) => props.name === selectedOption)
      : children[0];

    return (
      <div>
        {this.renderControls(selectedOption || children[0].props.name)}
        <div>{selectedChild}</div>
      </div>
    );
  }
}
