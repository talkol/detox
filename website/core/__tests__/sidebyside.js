import React from "react";
import SideBySide, { SideBySideOption } from "../SideBySide";
import renderer from "react-test-renderer";

describe("SideBySide", () => {
  it("should throw without children", () => {
    expect(<SideBySide />).toThrowErrorMatchingSnapshot();
  });

  it("should render first option if none is selected", () => {
    expect(
      renderer
        .create(
          <SideBySide>
            <SideBySideOption name="ios">iOS Content</SideBySideOption>
            <SideBySideOption name="android">Android Content</SideBySideOption>
          </SideBySide>
        )
        .toJSON()
    ).toMatchSnapshot();
  });

  it("should render the option that was selected", () => {
    const element = renderer.create(
      <SideBySide>
        <SideBySideOption name="ios">iOS Content</SideBySideOption>
        <SideBySideOption name="android">Android Content</SideBySideOption>
      </SideBySide>
    );
    element.getInstance().selectOption("android");
    expect(element.toJSON()).toMatchSnapshot();
  });
});
