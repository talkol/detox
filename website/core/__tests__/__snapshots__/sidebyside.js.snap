// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBySide should render first option if none is selected 1`] = `
<div>
  <div
    className="side-by-side-controls"
  >
    <a
      onClick={[Function]}
    >
      ios
    </a>
    <a
      onClick={[Function]}
    >
      android
    </a>
  </div>
  <div>
    <div>
      iOS Content
    </div>
  </div>
</div>
`;

exports[`SideBySide should render the option that was selected 1`] = `
<div>
  <div
    className="side-by-side-controls"
  >
    <a
      onClick={[Function]}
    >
      ios
    </a>
    <a
      onClick={[Function]}
    >
      android
    </a>
  </div>
  <div>
    <div>
      Android Content
    </div>
  </div>
</div>
`;

exports[`SideBySide should throw without children 1`] = `"received is not a function"`;
